packages/client/src/utils/referenceApi.ts

```diff
@@ -147,48 +148,109 @@ async function loadLocations(): Promise<ILocationDataResponse> {
   }
 
   const response = await res.json()
-  return response.data
+  const locations = response.entry.reduce(
+    (accumulator: { [key: string]: ILocation }, entry: fhir.BundleEntry) => {
+      if (!entry.resource || !entry.resource.id) {
+        throw new Error('Resource in entry not valid')
+      }
+
+      accumulator[entry.resource.id] = generateLocationResource(
+        entry.resource as fhir.Location
+      )
+
+      return accumulator
+    },
+    {}
+  )
+
+  return locations
 }
 
-async function loadFacilities(): Promise<IFacilitiesDataResponse> {
-  const url = `${window.config.COUNTRY_CONFIG_URL}/facilities`
-  const res = await fetch(url, {
-    method: 'GET',
-    headers: {
-      Authorization: `Bearer ${getToken()}`
-    }
-  })
+function generateLocationAddress(
+  fhirAddress: fhir.Address | undefined
+): string {
+  let address = ''
 
-  if (res && res.status !== 200) {
-    throw Error(res.statusText)
+  if (fhirAddress) {
+    address = Object.values(fhirAddress)
+      .filter((x) => typeof x === 'string' && x.length > 0)
+      .join(', ')
```

Comment:
@euanmillar Remember we were talking about not putting in `District` and `State` names in the address? It looks like this piece of code requires it. And we will also need to put `LOCATION_LEVEL_3`, `LOCATION_LEVEL_4` & `LOCATION_LEVEL_5`(if provided) location names in this address field to generate the name properly